ibis/omniscidb/client.py

```diff
@@ -652,21 +705,31 @@ def _execute(self, query, results=True):
         if isinstance(query, (DDL, DML)):
             query = query.compile()
 
-        if self.execution_type == EXECUTION_TYPE_ICP:
-            execute = self.con.select_ipc
-        elif self.execution_type == EXECUTION_TYPE_ICP_GPU:
-            execute = self.con.select_ipc_gpu
-        else:
-            execute = self.con.cursor().execute
+        if ipc is None and gpu_device is None:
+            ipc = self.ipc
+            gpu_device = self.gpu_device
+
+        self._check_execution_type(ipc, gpu_device)
 
         cursor = (
             OmniSciDBGeoCursor
             if FULL_GEO_SUPPORTED
             else OmniSciDBDefaultCursor
         )
 
+        params = {}
+
+        if ipc in (None, False) and gpu_device is None:
+            execute = self.con.cursor().execute
+        elif ipc and gpu_device is None:
+            execute = self.con.select_ipc
+        elif gpu_device is not None:
+            params['device_id'] = gpu_device
+            execute = self.con.select_ipc_gpu
+            cursor = OmniSciDBGPUCursor
+
```

Comment:
what happens on else?

raise an error

Reply:
sounds good.

Reply:
actually I changed the last `elif` to `else`